File: android_client/app/src/main/java/com/zigzag/client_app/model/ImageDescription.java
package com.zigzag.client_app.model;

import java.util.Map;

public class ImageDescription {
    private final String subtitle;
    private final String description;
    private final String sourceUri;
    private final String originalImageUriPath;
    private final Map<ScreenConfig, PhotoData> imageData;

    public ImageDescription(String subtitle, String description, String sourceUri,
            String originalImageUriPath, Map<ScreenConfig, PhotoData> imageData) {
        this.subtitle = subtitle;
        this.description = description;
        this.sourceUri = sourceUri;
        this.originalImageUriPath = originalImageUriPath;
        this.imageData = imageData;
    }

    public String getSubtitle() {
        return subtitle;
    }

    public String getDescription() {
        return description;
    }

    public String getSourceUri() {
        return sourceUri;
    }

    public String getOriginalImageUriPath() {
        return originalImageUriPath;
    }

    public Map<ScreenConfig, PhotoData> getImageData() {
        return imageData;
    }

    public PhotoData getBestMatchingImageData() {
        ScreenConfig k1 = new ScreenConfig("800", 800);
        ScreenConfig k2 = new ScreenConfig("480", 480);
        PhotoData v1 = imageData.get(k1);

        if (v1 == null) {
            return imageData.get(k2);
        } else {
            return v1;
        }
    }
}

File: ios_client/ios_client/models/ImageDescription.swift
/**
 * ImageDescription.swift
 * Created by Liviu Coman on 3/24/15.
 * Copyright (c) 2015 LiviuComan. All rights reserved.
 * based on design by HoriaComan
 */

import UIKit

/**
 * This class describes an image
 */
class ImageDescription: NSObject {
    /**
     * private imageDescription class variables
     */
    private var subtitle: String! = nil
    private var imgDescription: String! = nil
    private var sourceUri: String! = nil
    private var originalPhotoUriPath: String! = nil
    private var imageData: [ScreenConfig: ImageData] = [:]
    
    /**
     * custom constructor
     */
    init(subtitle: String, imgDescription: String, sourceUri: String, originalPhotoUriPath: String,
        imageData: [ScreenConfig: ImageData])
    {
        super.init()
        self.subtitle = subtitle
        self.imgDescription = imgDescription
        self.sourceUri = sourceUri
        self.originalPhotoUriPath = originalPhotoUriPath
        self.imageData = imageData
    }
    
    /**
     * custom getters
     */
    func getSubtitle() -> String{
        return self.subtitle
    }
    
    func getImgDescription() -> String {
        return self.imgDescription
    }
    
    func getSourceUri() -> String {
        return self.sourceUri
    }
    
    func getOriginalPhotoUriPath() -> String {
        return self.originalPhotoUriPath
    }
    
    func getImageData() -> [ScreenConfig: ImageData] {
        return self.imageData
    }
}
